src/atmega2560p/com/usart_initialize.rs

```diff
@@ -0,0 +1,491 @@
+    pub ucsrb: Volatile<u8>,
+    pub ucsrc: Volatile<u8>,
+    _pad: Volatile<u8>, // Padding to look for empty memory space.
+    pub ubrrl: Volatile<u8>,
+    pub ubrrh: Volatile<u8>,
+    pub udr: Volatile<u8>,
+}
+
+/// USART pointers declaration.
+impl Usart {
+    /// This creates a new memory mapped structure of the type USART for it's control.
+    pub unsafe fn new(num: UsartNum) -> &'static mut Usart {
+        match num {
+            UsartNum::Usart0 => &mut *(0xC0 as *mut Usart),
+            UsartNum::Usart1 => &mut *(0xC8 as *mut Usart),
+            UsartNum::Usart2 => &mut *(0xD0 as *mut Usart),
+            UsartNum::Usart3 => &mut *(0x130 as *mut Usart),
+        }
+    }
+}
+
+/// Various implementation functions for the USART protocol.
+impl Usart {
+    /// Function to disable global interrupts for smooth non-interrupted functioning of USART.
+    fn disable(&mut self) {
+        unsafe {
+            // Disable global interrupts.
+            interrupts::GlobalInterrupts::disable(&mut interrupts::GlobalInterrupts::new());
+        }
+    }
+
+    /// Function to re-enable global interrupts.
+    fn enable(&mut self) {
+        unsafe {
+            // Enable global interrupts.
+            interrupts::GlobalInterrupts::enable(&mut interrupts::GlobalInterrupts::new());
+        }
+    }
+
+    /// This function will return the Number of the USART according to the address.
+    fn get_num(&mut self) -> UsartNum {
+        let address = (self as *const Usart) as usize; // Gets address of usart structure.
+        match address {
+            // Return the number of USART used based on the address read.
+            0xC0 => UsartNum::Usart0,
+            0xC8 => UsartNum::Usart1,
+            0xD0 => UsartNum::Usart2,
+            0x130 => UsartNum::Usart3,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Function to get the port containing bits to
+    /// manipulate Recieve,Transmit and XCK bit of the particular USART.
+    fn get_port_xck(&mut self) -> (&mut port::Port, u8) {
+        let num: UsartNum = self.get_num();
+        match num {
+            UsartNum::Usart0 => (port::Port::new(port::PortName::E), USART0_XCK),
+            UsartNum::Usart1 => (port::Port::new(port::PortName::D), USART1_XCK),
+            UsartNum::Usart2 => (port::Port::new(port::PortName::H), USART2_XCK),
+            UsartNum::Usart3 => (port::Port::new(port::PortName::J), USART3_XCK),
+        }
+    }
+
+    /// Function to check the mode of the USART.
+    /// Returns 0 for asynchronous and 1 for synchronous.
+    fn get_mode(&mut self) -> bool {
+        let mut src = self.ucsrc.read();
+        src = src & (1 << 6);
+        if src == 0 {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /// Function to set the clock polarity mode which is of use in the
+    /// recieve and transmission implementation of USART.
+    pub fn set_polarity(&mut self, mode: UsartPolarity) {
```

Comment:
Don't use `Function to do  x,y,z` , instead write, `Does x,y,z` , the user knows this is a function.

Reply:
fixed everywhere.